### DISCUSSION TITLE
Merge badges from multiple emails.

### DISCUSSION THREAD

**Discussion by TrevorBenson** [community/community#35710] [Other Feature Feedback, Questions, & Ideas] [answered]
I used the [Merging Multiple Personal Accounts](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-your-personal-account/merging-multiple-personal-accounts) to merge my work email into my personal account. 

I also went through the list of [Why are my contributions not showing up on my profile?](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/managing-contribution-settings-on-your-profile/why-are-my-contributions-not-showing-up-on-my-profile) and starred to all the work repositories which my work email had made commits in.

I also restored my GPG and SSH keys and have SSO authorized the SSH. I'm only seeing the badges from the personal account, is this expected behavior? Does the entire commit graph need to finish rebuilding first? Will I just have to re-earn any badges which I no longer have?

I did notice that after starring all the private repositories of the organization there are still ~400 attributions missing even when I'm signed into the Orgs SSO. Signing into the Org SSO has a minimal impact on total contributions. I recall that combined I was around 650-750 contributions for the year with a bit below/above 200 in personal and 400+ in the Org. 

Now its ~240 total:

243 Not logged in / Incognito looking at profile:

![Screenshot from 2022-10-10 13-47-33](https://user-images.githubusercontent.com/9167887/194950246-65b930f8-d12d-4057-a48f-c53c86c71e37.png)

242 Logged into account, but not Org SSO:

![Screenshot from 2022-10-10 13-47-04](https://user-images.githubusercontent.com/9167887/194950154-bd63bf6b-c010-4c37-9e1b-8dafa30f73ec.png)

243 Logged into account and Authenticated to SSO:

![Screenshot from 2022-10-10 13-47-21](https://user-images.githubusercontent.com/9167887/194950198-d14e01bb-324b-45b3-9b24-1a227ef48773.png)

My suspicion is the missing contributions are why badges are missing. Are there any suggestions on investigating missing contributions? Or maybe tracking down all contributions for a deleted/ghost account to try and compare which ones are not in my graph?

**Comment by jfullstackdev** [comment]
badges are linked to email accounts, did you add all of them under one account ? but I'm not very sure if, say, you have both and it will be duplicated on the account you have now

also, as I was checking your profile, I see this

![image](https://user-images.githubusercontent.com/47092464/195134741-18cb937c-a947-4578-8b21-636569ea3ba5.png)

your last public activity was 2021 and they are your repos, most of those in your Repos tab are forks and you already read the rules for that because you shared the link, 

what else might you be missing ? oh I checked that again, they are just 2-3 commits missing ? I thought you are losing, say 100 commits difference

**Reply by TrevorBenson** [reply]
> badges are linked to email accounts, did you add all of them under one account ?

Yep. I had 2 accounts each with 1 email. Now both emails are attached and verified under this one account.

The missing commits went into private Org repos, not personal ones. I have starred each of those repos after merging/verifying the email to this account. The deleted account had maybe 5 personal repositories, but had minimal commits/merges. Maybe 10-20 contributions would be lost from allowing those personal repositories to be deleted.

> what else might you be missing ?

I don't know, that is why I am asking here. :)

- I did not use "hide my email" functionality. So do not believe the no reply email address which could hide contributions after moving accounts is related to this issue.
- I added the email, verified it, accepted the original Organizations invitation, signed into SSO. Also added the ssh and gpg, and added the SSO to the ssh key.
- I went to each private repository and added a :star: to each which is supposed to also restore the contributions. I'm also a member of the organization which owns the repositories and, obviously, previously submitted issues/pr's.

> they are just 2-3 commits missing ?

No, hundreds.  There were 350-450 commits in the work account before deletion. I recall reviewing 2022 commits for both accounts last week and the combined total was around 650-700 and I presumed 800-1000 would be my end of year totals.

For comparison I just went and unstarred 6 of the Org private repositories. I worked on each of these repositories in the last 3 months, and definitely many many times this year. There should be at least 100 commits related to these repositories, if not at least 200-300. My graph still shows I have 243 commits for 2022, identical to the number of contributions before unstarring these repos. If I choose to expand my **Contribution Activity** the Organizations repositories do not showup at all, as if I had never contributed in any way except from my personal email address.

**Comment by TrevorBenson** [comment]
I picked one of the smaller repositories and reviewed the history. I can see a merge/commit into the default branch from July 21st of this year, and 7 contributions for this year.

![Screenshot from 2022-10-11 10-21-40](https://user-images.githubusercontent.com/9167887/195158457-4d3550f7-ee60-402d-8283-9745785a8b85.png)

I double checked the commit `.patch` URL as well and this confirms the work email is definitely linked to this specific commit. However under the contribution graph July 21st says 0 contributions.

My total contributions say 243. I remove the star from this repository and 5 others which would have much greater contributions than this example. 10 minutes later my contributions still say 243. So even though it has linked past contributions to my current account which has the email attached, they do not appear to be included in contribution graphs regardless of adding a :star: to the repository or not and I can clearly see things attributed to my current account for specific dates are showing as 0 for that same date in my profile.

**Comment by TrevorBenson** [comment, marked as answer]
Solved, got some of my contributions back, but just a fraction. I gave examples of matching the contribution requirements still and of a url, .patch and missing dates in the graph to github support, they recalculated contributions for those repositories. However I was still missing over 1/2 of the contributions.


It turns out when the [merging accounts page](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-your-personal-account/merging-multiple-personal-accounts) stated:
> To attribute past commits to the new account, add the email address you used to author the commits to the account you're keeping. For more information, see "[Why are my contributions not showing up on my profile](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/managing-contribution-settings-on-your-profile/why-are-my-contributions-not-showing-up-on-my-profile#contributions-that-are-counted)?"

One **should** infer (although I did not at the time) that Pull Requests, Issues and Discussions will **not** be attributed to the new account. Unfortunately since this was not explicitly stated in either page I presumed that I could get back all contributions. If this was more clearly stated I might have decided initially to not create two accounts and later try to merge them. 

I hope this discussion helps others better understand the ramifications of merging accounts on past contributions.

**Comment by TrevorBenson** [comment]
> badges are linked to email accounts, did you add all of them under one account ? but I'm not very sure if, say, you have both and it will be duplicated on the account you have now
> 
> also, as I was checking your profile, I see this
> 
> ![image](https://user-images.githubusercontent.com/47092464/195134741-18cb937c-a947-4578-8b21-636569ea3ba5.png)
> 
> your last public activity was 2021 and they are your repos, most of those in your Repos tab are forks and you already read the rules for that because you shared the link, 
> 
> what else might you be missing ? oh I checked that again, they are just 2-3 commits missing ? I thought you are losing, say 100 commits difference

- There were only ever two emails. They are now under the same account. 
- There are hundreds of missing commits.

**Comment by muhammadsadoon** [comment]
Thanks for sharing the detailed explanation. I came across this discussion while reviewing contribution history issues, and the clarification about how GitHub counts contributions was very helpful.
